src/check-subscriptions.ts

```diff
@@ -0,0 +1,105 @@
+          {
+            type: 'bytes32',
+            name: '_conditionFunctionId',
+            value: ethers.utils.defaultAbiCoder.encode(
+              ['bytes4'],
+              [DapiServerInteface.getSighash('conditionPspBeaconUpdate')]
+            ),
+          },
+          { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
+        ]);
+
+        const chainId = chainNameToChainId[chainName];
+        if (!chainId) throw new Error(`🛑 Unknown chain name: ${chainName}`);
+
+        if (!credentials.networks[chainName].url) throw new Error(`🛑 No public RPC URL for chain ${chainName}`);
+        const provider = new ethers.providers.JsonRpcProvider(credentials.networks[chainName].url);
+
+        const DapiServerAddress = operationsRepository.chains[chainName].contracts.DapiServer;
+        if (!DapiServerAddress) throw new Error(`🛑 No DapiServer contract address for chain ${chainName}`);
+        const DapiServer = DapiServerContract(DapiServerAddress, provider);
+
+        const sponsor = chain.sponsor;
+
+        const expectedSubscriptionId = ethers.utils.keccak256(
+          ethers.utils.defaultAbiCoder.encode(
+            ['uint256', 'address', 'bytes32', 'bytes', 'bytes', 'address', 'address', 'address', 'bytes4'],
+            [
+              chainId,
+              airnodeAddress,
+              templateId,
+              parameters,
+              encodedBeaconUpdateSubscriptionConditions,
+              airnodeAddress,
+              sponsor,
+              DapiServerAddress,
+              DapiServerInteface.getSighash('fulfillPspBeaconUpdate'),
+            ]
+          )
+        );
+
+        const expectedBeaconId = ethers.utils.keccak256(
+          ethers.utils.solidityPack(['address', 'bytes32'], [airnodeAddress, templateId])
+        );
+        try {
+          console.log(`🔎 checking if subscriptionId ${expectedSubscriptionId} already exists for chain ${chainName}`);
+
+          const beaconId = await DapiServer.subscriptionIdToBeaconId(expectedSubscriptionId);
+
+          if (beaconId !== expectedBeaconId) {
+            throw new Error(`🛑 The subscription ID ${expectedSubscriptionId} does not exist`);
+          }
+
+          if (beaconId === expectedBeaconId)
+            return `✅ subscriptionId ${expectedSubscriptionId} exists for chain ${chainName}`;
+        } catch (error) {
+          console.error(error);
+          throw new Error(`🛑 Error checking subscription for beacon ${beaconName} on chain ${chainName}`);
+        }
+      })
+  );
+
+  const results = await Promise.allSettled(subscriptionPromises);
+
+  results.forEach((result) => {
+    if (result.status === 'fulfilled') {
+      console.log(result.value);
+    } else {
+      console.log(result.reason);
+    }
+  });
+
+  if (results.some((result) => result.status !== 'fulfilled')) throw new Error('🛑 Some subscription checks failed');
```

Comment:
Nit: you could maybe add a flag and set it to `true` in the else block of the `results.forEach` block above to avoid looping thru the results array again but I don't mind keeping it like it is now.

Reply:
yeah I think I need to do this in a better way, ill think of something cleaner